### PR TITLE
Fix all PVS V607-->V1423 HIGH level warnings

### PR DESCRIPTION
**Pull request by vadymeng0**
This commit is about to fix all the HIGH level warnings that was report by PVS. Most of the MEDIUM warnings were fixed also in this patch.
Please note that only warnings in Pastel code, and 3rd party code will be fixed. Dependencies code were left unfixed.
Details can be referred in below table.
Capture

Result: 
    - Built successfully
    - PVS report show no warnings for all the issues on column (Pastel (Fixed by this commit))
    - Passed all circleci test cases
    - Passed all below test suite on local machine:
              cd qa/test-suite
              ./full_test_suite.py btest
              ./full_test_suite.py gtest
              ./full_test_suite.py sec-hard
              ./full_test_suite.py no-dot-so
              ./full_test_suite.py util-test
              ./full_test_suite.py secp256k1
              ./full_test_suite.py libsnark
              ./full_test_suite.py univalue
              ./full_test_suite.py rpc-common
              ./full_test_suite.py rpc-ext
   - Test case that took too long to complete, and failed some test cases while running:
              ./full_test_suite.py rpc-mn

### CODE REVIEW

**Review by a-ok123 — APPROVED** [review]

File: src/secp256k1/src/field_10x26_impl.h [hunk 1]
```diff
@@ -809,11 +809,11 @@ SECP256K1_INLINE static void secp256k1_fe_sqr_inner(uint32_t *r, const uint32_t
...
-       + (uint64_t)(a[1]*2) * a[8]
-       + (uint64_t)(a[2]*2) * a[7]
-       + (uint64_t)(a[3]*2) * a[6]
-       + (uint64_t)(a[4]*2) * a[5];
+    d  = ( (uint64_t)a[0]*2 ) * a[9]
+       + ( (uint64_t)a[1]*2 ) * a[8]
+       + ( (uint64_t)a[2]*2 ) * a[7]
+       + ( (uint64_t)a[3]*2 ) * a[6]
+       + ( (uint64_t)a[4]*2 ) * a[5];
     /* VERIFY_BITS(d, 64); */
```
Comment (a-ok123) [inline src/secp256k1/src/field_10x26_impl.h:817]: I wonder if this will change the results of key generation and/or signing/verification

**Review by a-ok123 — CHANGES_REQUESTED** [review]
can you please move changes to secp256 arithmetic to separate PR?

File: src/secp256k1/src/field_10x26_impl.h (hunk 1 above)
Comment (Dicklesworthstone) [reply src/secp256k1/src/field_10x26_impl.h]: We should add extra tests for this if it's even a potential risk.

Comment (vadymeng0) [reply src/secp256k1/src/field_10x26_impl.h]: secp256 arithmetic changes has been moved to ticket: https://github.com/pastelnetwork/pastel/pull/24

**Comment by vadymeng0** [comment]
secp256 arithmetic changes has been moved to ticket: #24

**Review by a-ok123 — APPROVED** [review]

**Pull request merged**